internal/commands/vaultsecrets/secrets/create.go

```diff
@@ -439,9 +527,124 @@ func validateSecretConfig(secretConfig SecretConfig) []string {
 		missingKeys = append(missingKeys, "type")
 	}
 
-	if secretConfig.IntegrationName == "" {
-		missingKeys = append(missingKeys, "integration_name")
+	return missingKeys
+}
+
+var availableRotatingSecretProviders = map[string]map[string]any{
+	string(integrations.Twilio):       twilioRotatingSecretConfig,
+	string(integrations.MongoDBAtlas): mongoDBAtlasRotatingSecretConfig,
+	string(integrations.AWS):          awsRotatingSecretConfig,
+	string(integrations.GCP):          gcpRotatingSecretConfig,
+}
+
+var availableDynamicSecretProviders = map[string]map[string]any{
+	string(integrations.AWS): awsDynamicSecretConfig,
+	string(integrations.GCP): gcpDynamicSecretConfig,
+}
+
+func promptUserForConfig(opts *CreateOpts) (SecretConfig, secretConfigInternal, error) {
+	var (
+		config         SecretConfig
+		internalConfig secretConfigInternal
+		err            error
+		providerFields map[string]map[string]any
+		items          []string
+	)
+
+	if !opts.IO.CanPrompt() {
+		return config, internalConfig, fmt.Errorf("unable to create secret interactively")
 	}
 
-	return missingKeys
+	if opts.Type == secretTypeDynamic {
+		providerFields = availableDynamicSecretProviders
+		items = maps.Keys(providerFields)
+	} else if opts.Type == secretTypeRotating {
+		providerFields = availableRotatingSecretProviders
+		items = maps.Keys(providerFields)
+	}
+
+	providerPrompt := promptui.Select{
+		Label:  "Please select the provider you would like to configure",
+		Items:  items,
+		Stdin:  io.NopCloser(opts.IO.In()),
+		Stdout: iostreams.NopWriteCloser(opts.IO.Err()),
+	}
+
+	_, provider, err := providerPrompt.Run()
+	if err != nil {
+		return config, internalConfig, fmt.Errorf("provider selection prompt failed: %w", err)
+	}
+	config.Type = integrations.IntegrationType(provider)
+
+	fieldValues, err := populateFieldValues(providerFields[provider], opts)
+	if err != nil {
+		return config, internalConfig, err
+	}
+
+	internalConfig.Details = fieldValues
+	return config, internalConfig, err
+}
+
+func populateFieldValues(fieldMap map[string]any, opts *CreateOpts) (map[string]any, error) {
+	var fieldPrompt promptui.Prompt
+	fieldsMap := make(map[string]any)
```

Comment:
Nit: `fieldMap` and `fieldsMap` are a bit too similar and could get confused within the same function